contracts/src/v0.8/ccip/test/pools/GHO/fork/GhoTokenPoolMigrate1_4To1_5/TokenPoolsUpgrade.t.sol

```diff
@@ -0,0 +1,239 @@
+  }
+
+  function testLockOrBurnViaRouter() public {
+    uint256 amount = 10e18;
+    Client.EVM2AnyMessage memory message = _generateMessage(alice, 1);
+
+    {
+      vm.selectFork(l1.forkId);
+
+      message.tokenAmounts[0] = Client.EVMTokenAmount({token: address(l1.token), amount: amount});
+      uint256 feeTokenAmount = l1.router.getFee(l2.chainSelector, message);
+
+      // router uses 1_5 onRamp
+      assertEq(l1.router.getOnRamp(l2.chainSelector), address(l1.EVM2EVMOnRamp1_5));
+      vm.expectEmit();
+      emit CCIPSendRequested(_messageToEvent(message, l1.EVM2EVMOnRamp1_5, feeTokenAmount, alice, true));
+      vm.prank(alice);
+      l1.router.ccipSend{value: feeTokenAmount}(l2.chainSelector, message);
+    }
+
+    {
+      vm.selectFork(l2.forkId);
+
+      message.tokenAmounts[0] = Client.EVMTokenAmount({token: address(l2.token), amount: amount});
+      uint256 feeTokenAmount = l2.router.getFee(l1.chainSelector, message);
+
+      // router uses 1_5 onRamp
+      assertEq(l2.router.getOnRamp(l1.chainSelector), address(l2.EVM2EVMOnRamp1_5));
+      vm.expectEmit();
+      emit CCIPSendRequested(_messageToEvent(message, l2.EVM2EVMOnRamp1_5, feeTokenAmount, alice, false));
+      vm.prank(alice);
+      l2.router.ccipSend{value: feeTokenAmount}(l1.chainSelector, message);
+    }
+  }
+
+  function testRevertLockOrBurnVia1_2OnRamp() public {
+    uint256 amount = 10e18;
+    {
+      vm.selectFork(l1.forkId);
+      vm.expectRevert(abi.encodeWithSelector(CallerIsNotARampOnRouter.selector, address(l1.EVM2EVMOnRamp1_2)));
+      vm.prank(address(l1.EVM2EVMOnRamp1_2));
+      l1.tokenPool.lockOrBurn(alice, abi.encode(alice), amount, l2.chainSelector, "");
+    }
+    {
+      vm.selectFork(l2.forkId);
+      vm.expectRevert(abi.encodeWithSelector(CallerIsNotARampOnRouter.selector, address(l2.EVM2EVMOnRamp1_2)));
+      vm.prank(address(l2.EVM2EVMOnRamp1_2));
+      l2.tokenPool.lockOrBurn(alice, abi.encode(alice), amount, l1.chainSelector, "");
+    }
+  }
+
+  function testReleaseOrMintVia1_2OffRamp() public {
+    uint256 amount = 10e18;
+    Client.EVM2AnyMessage memory message = _generateMessage(alice, 1);
+
+    // off ramp on L1
+    {
+      // build message
+      vm.selectFork(l2.forkId);
+      message.tokenAmounts[0] = Client.EVMTokenAmount({token: address(l2.token), amount: amount});
+      uint256 feeTokenAmount = l2.router.getFee(l1.chainSelector, message);
+      Internal.EVM2EVMMessage memory eventArg = _messageToEvent(
+        message,
+        l2.EVM2EVMOnRamp1_5,
+        feeTokenAmount,
+        alice,
+        false
+      );
+
+      // test off ramp
+      vm.selectFork(l1.forkId);
+
+      uint256 balanceBefore = l1.token.balanceOf(alice);
+
+      vm.expectEmit(address(l1.tokenPool));
+      emit Released(address(l1.EVM2EVMOffRamp1_2), alice, amount);
+      vm.prank(address(l1.EVM2EVMOffRamp1_2));
+      l1.EVM2EVMOffRamp1_2.executeSingleMessage(eventArg, new bytes[](message.tokenAmounts.length));
```

Comment:
1.2 offramp still works after the cutover but 1.2 onramp does not, right?
this is because old messages must be executable after the upgrade right

Reply:
yes that's correct